packages/core-js/modules/web.structured-clone.js

```diff
@@ -0,0 +1,409 @@
+      } else throwUncloneableType(type);
+      break;
+    default:
+      if (nativeRestrictedStructuredClone) {
+        cloned = nativeRestrictedStructuredClone(value);
+      } else switch (type) {
+        case 'BigInt':
+          // can be a 3rd party polyfill
+          cloned = Object(value.valueOf());
+          break;
+        case 'Boolean':
+          cloned = Object(bolleanValueOf(value));
+          break;
+        case 'Number':
+          cloned = Object(numberValueOf(value));
+          break;
+        case 'String':
+          cloned = Object(stringValueOf(value));
+          break;
+        case 'Date':
+          cloned = new Date(getTime(value));
+          break;
+        case 'RegExp':
+          cloned = new RegExp(value);
+          break;
+        case 'ArrayBuffer':
+          // detached buffers throws on `.slice`
+          try {
+            cloned = value.slice(0);
+          } catch (error) {
+            throw new DOMException('ArrayBuffer is deatched', DATA_CLONE_ERROR);
+          } break;
+        case 'SharedArrayBuffer':
+          // SharedArrayBuffer should use shared memory, we can't polyfill it, so return the original
+          cloned = value;
+          break;
+        case 'Blob':
+          try {
+            cloned = value.slice(0, value.size, value.type);
+          } catch (error) {
+            throwUncloneableType(type);
+          } break;
+        case 'DOMPoint':
+        case 'DOMPointReadOnly':
+          C = global[type];
+          try {
+            cloned = C.fromPoint ? C.fromPoint(value) : new C(value.x, value.y, value.z, value.w);
+          } catch (error) {
+            throwUncloneableType(type);
+          } break;
+        case 'DOMRect':
+        case 'DOMRectReadOnly':
+          C = global[type];
+          try {
+            cloned = C.fromRect ? C.fromRect(value) : new C(value.x, value.y, value.width, value.height);
+          } catch (error) {
+            throwUncloneableType(type);
+          } break;
+        case 'DOMMatrix':
+        case 'DOMMatrixReadOnly':
+          C = global[type];
+          try {
+            cloned = C.fromMatrix ? C.fromMatrix(value) : new C(value);
+          } catch (error) {
+            throwUncloneableType(type);
+          } break;
+        case 'AudioData':
+        case 'VideoFrame':
+          try {
+            cloned = value.clone();
+          } catch (error) {
+            throwUncloneableType(type);
+          } break;
+        case 'File':
+          try {
+            cloned = new File([value], value.name, value);
+          } catch (error) {
+            throwUncloneableType(type);
+          } break;
+        default:
+          throwUncloneableType(type);
+      }
+  }
+
+  mapSet(map, value, cloned);
+
+  if (deep) switch (type) {
+    case 'Array':
+    case 'Object':
+      for (key in value) if (hasOwn(value, key)) {
+        cloned[key] = structuredCloneInternal(value[key], map);
```

Comment:
It seems that `CreateDataProperty` (./internals/create-property) should be used here.

Reply:
Makes sense, thanks.